src/azul/service/storage_service.py

```diff
@@ -153,13 +161,22 @@ def get_presigned_url(self, key: str, file_name: Optional[str] = None) -> str:
                           request to the signed URL. If None, no such header will be present in the response.
         """
         assert file_name is None or '"' not in file_name
+        url_lifespan_in_seconds = 1 * 60 * 60
+        now = datetime.now(tz=tzutc())
+        credentials_expiration = self.client._request_signer._credentials._expiry_time  # type: ignore
```

Comment:
```suggestion
        expiry_time = self.client._request_signer._credentials._expiry_time  # type: ignore
```

Reply:
Is there no alternative that avoids accessing the private attributes?

Reply:
We could potentially use `_seconds_remaining` from the `RefreshableCredentials` class, however that's a private method, and we'd have to further modify our dummy credentials in order to make them compatible.
Also, accessing the expiration value [this way](https://github.com/DataBiosphere/azul/issues/5503#issuecomment-1724913389) was specified by Hannes in his original design.